### PR TITLE
Add basic media element support

### PR DESCRIPTION
**Pull request by RebeccaAnne**
This implements the basic handling for media elements. There's still a bunch of TODOs coming in future changes, but this gets us parsing, object model, and basic rendering.

#1354

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -25,6 +25,10 @@
 		6B1147D81F32F922008846EC /* ACRShowCardTarget.h in Headers */ = {isa = PBXBuildFile; fileRef = 6B1147D61F32F922008846EC /* ACRShowCardTarget.h */; };
 		6B1147D91F32F922008846EC /* ACRShowCardTarget.mm in Sources */ = {isa = PBXBuildFile; fileRef = 6B1147D71F32F922008846EC /* ACRShowCardTarget.mm */; };
 		6B6840F91F25EC2D008A933F /* ACRInputChoiceSetRenderer.mm in Sources */ = {isa = PBXBuildFile; fileRef = 6B6840F71F25EC2D008A933F /* ACRInputChoiceSetRenderer.mm */; };
+		6B7B1A9120B4D2AB00260731 /* Media.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 6B7B1A8D20B4D2AA00260731 /* Media.cpp */; };
+		6B7B1A9220B4D2AB00260731 /* MediaSource.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 6B7B1A8E20B4D2AA00260731 /* MediaSource.cpp */; };
+		6B7B1A9320B4D2AB00260731 /* MediaSource.h in Headers */ = {isa = PBXBuildFile; fileRef = 6B7B1A8F20B4D2AA00260731 /* MediaSource.h */; };
+		6B7B1A9420B4D2AB00260731 /* Media.h in Headers */ = {isa = PBXBuildFile; fileRef = 6B7B1A9020B4D2AB00260731 /* Media.h */; };
 		6BF339D320A6649500DA5973 /* json.h in CopyFiles */ = {isa = PBXBuildFile; fileRef = F4071C771FCCBAEF00AF4FEA /* json.h */; };
```
Comment (RebeccaAnne) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards.xcodeproj/project.pbxproj]: Thanks @nesalang!

File: source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl [hunk 2]
```diff
@@ -577,6 +580,76 @@ AdaptiveNamespaceStart
         interface IAdaptiveCardElement;
     };
 
+    [
+        version(NTDDI_WIN10_RS1),
+#ifdef ADAPTIVE_CARDS_WINDOWS
+        uuid(05F2F77A-D458-4F73-AAF7-06B91D4C37E6),
+        contract(InternalContract, 1),
+        internal
```
Comment (khouzam) [inline source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl]: Should this be exclusiveto?

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 3]
```diff
@@ -2768,6 +2750,120 @@ AdaptiveNamespaceStart
...
+
+        // If the host doesn't support interactivity we're done here, just return the poster image
+        if (!SupportsInteractivity(hostConfig.Get()))
+        {
+            renderContext->AddWarning(
+                ABI::AdaptiveNamespace::WarningStatusCode::InteractivityNotSupported,
+                HStringReference(L"Media was present in card, but interactivity is not supported").Get());
+
+            THROW_IF_FAILED(posterImage.CopyTo(&mediaControl));
+            return;
```
Comment (khouzam) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: If there's no poster image, on interactive cards we'll use the media element, but if interactivity is not supported, we'll fail?

File: source/uwp/Renderer/lib/XamlBuilder.cpp [hunk 4]
```diff
@@ -2768,6 +2750,120 @@ AdaptiveNamespaceStart
...
+            THROW_IF_FAILED(touchTargetAsButtonBase->add_Click(Callback<IRoutedEventHandler>([posterContainer, mediaElement](IInspectable* /*sender*/, IRoutedEventArgs* /*args*/) -> HRESULT
+            {
+                THROW_IF_FAILED(posterContainer->put_Visibility(Visibility_Collapsed));
+
+                ComPtr<IUIElement> mediaAsUIElement;
+                THROW_IF_FAILED(mediaElement.As(&mediaAsUIElement));
+                THROW_IF_FAILED(mediaAsUIElement->put_Visibility(Visibility_Visible));
+
+                THROW_IF_FAILED(mediaElement->Play());
+
```
Comment (khouzam) [inline source/uwp/Renderer/lib/XamlBuilder.cpp]: Is throwing from an EventHandler the right thing to do?

File: source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl (hunk 2 above)
Comment (RebeccaAnne) [reply source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl]: Yes, fixed.

---
In reply to: [190355696](https://github.com/Microsoft/AdaptiveCards/pull/1490#discussion_r190355696) [](ancestors = 190355696)

File: source/uwp/Renderer/lib/XamlBuilder.cpp (hunk 4 above)
Comment (RebeccaAnne) [reply source/uwp/Renderer/lib/XamlBuilder.cpp]: Nope. Switched to RETURN_IF_FAILED.

---
In reply to: [190370228](https://github.com/Microsoft/AdaptiveCards/pull/1490#discussion_r190370228) [](ancestors = 190370228)

**Pull request merged**